From 0b91acacd7caa1d85370c63096561cb23afe63c5 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 12 Feb 2025 12:20:35 +0100
Subject: [PATCH 1488/1695] dt-bindings: add the DBGMCU secure debug mailbox
 bindings

Document DBMCU secure mailbox bindings.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4b323c57d883e47a5c6cad445c0f8fd3db30712e dt-bindings: add the DBGMCU secure debug mailbox bindings]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: Id2bcf8cda3f5b71ecb47dca11fd8866f570958fe
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/436472
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../soc/stm32/st,stm32mp21-dbgmcu-mbx.yaml    | 55 +++++++++++++++++++
 1 file changed, 55 insertions(+)
 create mode 100644 documentation/devicetree/bindings/soc/stm32/st,stm32mp21-dbgmcu-mbx.yaml

diff --git a/documentation/devicetree/bindings/soc/stm32/st,stm32mp21-dbgmcu-mbx.yaml b/documentation/devicetree/bindings/soc/stm32/st,stm32mp21-dbgmcu-mbx.yaml
new file mode 100644
index 000000000..4c818723a
--- /dev/null
+++ b/documentation/devicetree/bindings/soc/stm32/st,stm32mp21-dbgmcu-mbx.yaml
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/soc/stm32/st,stm32mp21-dbgmcu-mbx.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STM32 DBGMCU secure debug mailbox for Authenticated Debug Access Control
+
+maintainers:
+  - Amelie Delaunay <amelie.delaunay@foss.st.com>
+
+description: |
+  On STM32MP21, the debug authentication protocol implements a bidirectional
+  communication between the debugger (host) and the target (STM32MP21) through
+  a mailbox interface located in the DBGMCU.
+
+properties:
+  compatible:
+    const: st,stm32mp21-dbgmcu-mbx
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  interrupts:
+    items:
+      - description: Detection of debugger write in DBG_AUTH_HOST register
+      - description: Detection of debugger read DBG_AUTH_DEV register
+
+  interrupt-names:
+    items:
+      - const: dbg_auth_wr
+      - const: dbg_auth_rd
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - interrupts
+  - interrupt-names
+
+additionalProperties: false
+
+examples:
+  - |
+        dbgmcu_mbx: mailbox@440a0100 {
+          compatible = "st,stm32mp21-dbgmcu-mbx";
+          reg = <0x440a0100 0xc>;
+          interrupts = <GIC_SPI 208 IRQ_TYPE_LEVEL_HIGH>,
+                       <GIC_SPI 209 IRQ_TYPE_EDGE_RISING>;
+          interrupt-names = "dbg_auth_wr", "dbg_auth_rd";
+          clocks = <&rcc CK_DBGMCU>;
+        };
-- 
2.47.2

